experimental/cache_size_test.cc

```diff
@@ -0,0 +1,49 @@
+
+#include "cache_size.h"
+
+#include <iostream>
+// Measure how often measuring time is able to accurately determine cache hits
+// and cache misses based on the buffer size under analysis and the cache size
+// it checks whether buffer sizes that fit in cache have less reading time or
+// not
+int main(int argc, char* argv[]) {
+  static constexpr long kCacheSize = 8 * 1024 * 1024;
+  static constexpr long kIterations = 2;
+  static constexpr long kTestSizes = 4;
+
+  // generates a list of numbers that are used to determine the size of buffers
+  std::vector<int> candidates(kTestSizes);
```

Comment:
nit: .at() isn't necessary if we know it will be within bounds. I'd probably change this to:

```c++
std::vector<int> candidates;
for (int i = 0; i < kTestSizes; ++i) {
  candidates.push_back(2 * (i + 1));
}
```

This has fewer opportunities for off-by-one errors.